libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.helpers.spec.ts

```diff
@@ -0,0 +1,265 @@
+      const result = formatDecimalPlaces(
+        new BigNumber('13.6898828123'),
+      ).toString();
+
+      expect(result).toEqual('13.689882');
+    });
+
+    it('should handle zero and negative numbers in formatDecimalPlaces', () => {
+      expect(formatDecimalPlaces(new BigNumber(0)).toString()).toBe('0');
+      expect(formatDecimalPlaces(new BigNumber('-1.2345678')).toString()).toBe(
+        '-1.234567',
+      );
+    });
+  });
+
+  describe('calculateCreditPercentage', () => {
+    it('should return the correct percentage that the participant has in the credit', () => {
+      const result = formatDecimalPlaces(
+        new BigNumber('13.6898828123'),
+      ).toString();
+
+      expect(result).toEqual('13.689882');
+    });
+
+    it('should return 0 if amountTotal is zero', () => {
+      const result = calculateCreditPercentage(
+        new BigNumber(10),
+        new BigNumber(0),
+      );
+
+      expect(result).toBe('0');
+    });
+
+    it('should return 0 if amount is zero', () => {
+      expect(
+        calculateCreditPercentage(new BigNumber(0), new BigNumber(100)),
+      ).toBe('0');
+    });
+  });
+
+  describe('calculateAmount', () => {
+    it('should return the correct amount of the participant', () => {
+      const amount = calculateAmount({
+        creditsDocumentUnitPrice,
+        massIdCertificateValue: new BigNumber(1500),
+        massIdPercentage: '30.1234567892',
+        participantAmount: '0',
+      });
+
+      expect(amount.toString()).toEqual('2363.185185');
+    });
+
+    it('should return the correct amount of the participant when the same participant already exists', () => {
+      const amount = calculateAmount({
+        creditsDocumentUnitPrice,
+        massIdCertificateValue: new BigNumber(1500),
+        massIdPercentage: '30.1234567892',
+        participantAmount: '3.223669',
+      });
+
+      expect(amount.toString()).toEqual('2366.408854');
+    });
+
+    it('should handle undefined participantAmount', () => {
+      const amount = calculateAmount({
+        creditsDocumentUnitPrice,
+        massIdCertificateValue: new BigNumber(100),
+        massIdPercentage: '10',
+        participantAmount: undefined,
+      });
+
+      expect(amount).toBe('52.3');
+    });
+  });
+
+  describe('calculateRemainder', () => {
+    it('should return the correct remainder of the credit', () => {
+      const massIdTotalValue = new BigNumber(2500);
+      const actors: ActorsByActorType = new Map();
+
+      // eslint-disable-next-line @typescript-eslint/naming-convention, no-underscore-dangle, @typescript-eslint/no-unused-vars
+      for (const _iterator of Array.from({ length: 3 })) {
+        actors.set(
+          `${random<RewardsDistributionActorType>()}-${faker.string.uuid()}`,
+          {
+            ...random<RewardsDistributionActor>(),
+            amount: '12.503982',
+            percentage: '33.333333',
+          },
+        );
+      }
```

Comment:
May using a array.map would be more elegant and avoid lint issues. Something like:

```suggestion
     const actors = new Map(Array.from({ length: 3 }).map(() => ([
       `${random<RewardsDistributionActorType>()}-${faker.string.uuid()}`,
       {
            ...random<RewardsDistributionActor>(),
            amount: '12.503982',
            percentage: '33.333333',
          },
      ])));
```